File: src/java/org/elastos/hive/vault/scripting/InsertExecutable.java
package org.elastos.hive.vault.scripting;

import com.fasterxml.jackson.databind.JsonNode;
import com.google.gson.annotations.SerializedName;

public class InsertExecutable extends Executable {
    public InsertExecutable(String name, String collectionName, JsonNode document, JsonNode options) {
        super(name, Type.INSERT, null);
        super.setBody(new Body(collectionName, document, options));
    }

    public InsertExecutable(String name, String collectionName, JsonNode document) {
        this(name, collectionName, document, null);
    }

    private class Body extends DatabaseBody {
        @SerializedName("document")
        private JsonNode document;
        @SerializedName("options")
        private JsonNode options;

        public Body(String collection, JsonNode document, JsonNode options) {
            super(collection);
            this.document = document;
            this.options = options;
        }
    }
}

File: src/service/scripting/insertexecutable.ts
import {ExecutableDatabaseBody, ExecutableType, Executable} from "./executable";
import {InvalidParameterException} from "../../exceptions";

export class InsertExecutableBody extends ExecutableDatabaseBody {
    constructor(collection: string, private document: any, private options?: any) {
        super(collection);

        if (!this.document)
            throw new InvalidParameterException('Invalid document');
    }

    getDocument(): any {
        return this.document;
    }

    getOptions(): any {
        return this.options;
    }
}

/**
 * Used to insert the documents.
 */
export class InsertExecutable extends Executable {
    constructor(name: string, collection: string, document: any, options: any) {
        super(name, ExecutableType.INSERT, new InsertExecutableBody(collection, document, options));
    }
}
